### PR TITLE
Allow use of 2 different language servers

### PR DESCRIPTION
**Pull request by huguesv**
Moved LS specific code out of the client and into new server class.

Configure which one is used via env var to allow testing either server. Defaults to the .NET Core server that is currently included in extension.

Configure the location of the node based server via env var since it is not included in the extension.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                return true;
+            }
+
+            return version >= PythonLanguageVersion.V30 || version == PythonLanguageVersion.None;
+        }
+
+        public async override Task<Connection> ActivateAsync() {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var nodePath = GetNodeExecutablePath();
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs]: ```nodePath ``` can be null

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        private string GetNodeExecutablePath() {
+            string filePath;
+
+            // Prefer the node executable from the server folder, if there is one
+            var serverFolderPath = GetServerLocation();
+            if (Directory.Exists(serverFolderPath)) {
+                filePath = Path.Combine(serverFolderPath, "node.exe");
+                if (File.Exists(filePath)) {
+                    return filePath;
+                }
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs]: do we care about logging errors?

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerDotNetCore.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            };
+
+            if (process.Start()) {
+                return new Connection(process.StandardOutput.BaseStream, process.StandardInput.BaseStream);
+            }
+
+            return null;
+        }
+
+        public override object CreateInitializationOptions(string interpreterPath, string interpreterVersion, string rootPath, IEnumerable<string> searchPaths) {
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerDotNetCore.cs]: should we add string checks?

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+            }
+
+            return val != "0" && string.Compare(val, "false", StringComparison.OrdinalIgnoreCase) != 0;
+        }
+
+        private string GetNodeExecutablePath() {
+            string filePath;
+
+            // Prefer the node executable from the server folder, if there is one
+            var serverFolderPath = GetServerLocation();
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs]: ```serverFolderPath``` can be null

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs (hunk 2 above)
Comment (huguesv) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs]: right now they'll get a message box if either node or the server is missing.

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs (hunk 1 above)
Comment (huguesv) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs]: null is accepted by File.Exists

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs (hunk 4 above)
Comment (huguesv) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerNodejs.cs]: Directory.Exists accepts null

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerDotNetCore.cs (hunk 3 above)
Comment (huguesv) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageServerDotNetCore.cs]: searchPaths was the only that could cause a crash, I've added a check for null.

**Pull request merged**